### PR TITLE
Release 6.7.0

### PR DESCRIPTION
**Pull request by cyattilakiss**
**Description**
#942 [PW-3388] Fix Payments request 
#947 Adding periods to 'The payment is REFUSED' messages 
#954 [PW-3707] create local version of isPaceOrderAllowed for HPP 
#955 [PW-3708] Update the quoteid for refuse payments 
#949 Added missing 'items' data required for checkout agreements 
#960 [PW-4027] Shopper reference is too short for api v66 
#961 [PW-3950] All the klarna variants skips the no required fields 
#967 Add mbway icon in v6 
#957 [PW-3873] New 3DS1 flow with Adyen's termUrl and redirectResult 
#959 [PW-3708] Cloning quote with quote helper on failed payment 
#972 [PW-4070] Using TaxAmount for open invoice line items instead of calculating the value
#980 Add extra logging for invoice creation
#981 Fix undefined variable 'details' when processing HPP recurring contract
#956 [PW-3771] Revise Recurring settings
#977 [PW-2089] Config to choose street line to be used as house number

**Tested scenarios**
E2E test cases are passing

**Fixed issue**:  
#873 [PW-3388] Credit cards are being saved without selecting the option 
#937 [PW-3707] Cannot place order with other than Adyen payment method
#915 [PW-3708] Duplicated Quote IDs in sales_order_table
#948 [PW-3880] Unable to check checkout agreements for alternative payment methods (HPP)

### CODE REVIEW

File: Helper/Requests.php [hunk 1]
```diff
@@ -357,6 +357,17 @@ public function buildThreeDS2Data($additionalData, $storeId, $request = [])
...
     }
 
+    /**
+     * @param array $request
+     * @return array
+     */
+    public function buildRedirectData($storeId, $request = [])
+    {
+        $request['returnUrl'] = rtrim($this->adyenHelper->getOrigin($storeId), '/') .
+            '/adyen/process/redirect';
```
Comment (cyattilakiss) [inline Helper/Requests.php:367]: @acampos1916 shouldn't this be the transparent controller?

Comment (acampos1916) [reply Helper/Requests.php:367]: @cyattilakiss will check!

Comment (acampos1916) [reply Helper/Requests.php:367]: The transparent controller handles POST parameters, and this returnUrl is used when the 3DS1 behavior returns a GET redirectResult. This redirectResult is handled in `/adyen/process/redirect`

**Review by AlexandrosMor — APPROVED** [review]

File: etc/di.xml [hunk 2]
```diff
@@ -560,6 +560,7 @@
                 <item name="transaction" xsi:type="string">Adyen\Payment\Gateway\Request\CcAuthorizationDataBuilder</item>
                 <item name="vault" xsi:type="string">Adyen\Payment\Gateway\Request\VaultDataBuilder</item>
                 <item name="threeds2" xsi:type="string">Adyen\Payment\Gateway\Request\ThreeDS2DataBuilder</item>
+                <item name="redirect" xsi:type="string">Adyen\Payment\Gateway\Request\RedirectDataBuilder</item>
```
Comment (Aleffio) [inline etc/di.xml:563]: if this includes some 3ds1 redirect logic, should this also apply to `AdyenPaymentOneclickAuthorizeRequest` for billing agreements?

**Review by msilvagarcia — DISMISSED** [review]

**Review by msilvagarcia — DISMISSED** [review]

**Review by AlexandrosMor — DISMISSED** [review]

**Review by msilvagarcia — APPROVED** [review]

File: etc/di.xml (hunk 2 above)
Comment (acampos1916) [reply etc/di.xml:562]: Thanks @Aleffio, indeed it is missing https://github.com/Adyen/adyen-magento2/pull/986

**Review by msilvagarcia — APPROVED** [review]

**Review by acampos1916 — DISMISSED** [review]

**Comment by AlexandrosMor** [comment]
When we release the 6.9.0 we can publish the new docs related to recurring settings

**Review by acampos1916 — APPROVED** [review]

**Review by msilvagarcia — APPROVED** [review]

**Pull request merged**